src/main/java/com/dutact/web/features/event/admin/services/EventServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.dutact.web.features.event.admin.services;
+
+import com.dutact.web.core.entities.EventOrganizer;
+import com.dutact.web.core.entities.event.Event;
+import com.dutact.web.core.repositories.EventRepository;
+import com.dutact.web.features.event.admin.dtos.EventCreateUpdateDto;
+import com.dutact.web.features.event.admin.dtos.EventDto;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+
+@Service("organizerEventService")
+public class EventServiceImpl implements EventService {
+    private final EventMapper eventMapper;
+    private final EventRepository eventRepository;
+
+    public EventServiceImpl(EventMapper eventMapper, EventRepository eventRepository) {
+        this.eventMapper = eventMapper;
+        this.eventRepository = eventRepository;
+    }
+
+    @Override
+    public EventDto createEvent(Integer orgId, EventCreateUpdateDto eventDto) {
+        Event event = eventMapper.toEvent(eventDto);
+        event.setOrganizer(new EventOrganizer(orgId));
+        eventRepository.save(event);
+
+        return eventMapper.toEventDto(event);
+    }
+
+    @Override
+    public Collection<EventDto> getEvents(Integer orgId) {
+        return eventRepository.findAllByOrganizerId(orgId)
+                .stream().map(eventMapper::toEventDto).toList();
+    }
+
+    @Override
+    public EventDto getEvent(Integer eventId) {
+        return eventMapper.toEventDto(eventRepository.findById(eventId).orElseThrow());
+    }
+
+    @Override
+    public EventDto updateEvent(Integer eventId, EventCreateUpdateDto eventDto) {
+        Event event = eventRepository.findById(eventId).orElseThrow();
```

Comment:
Don't  we need to save update history

Reply:
Not in scope, do later 
![image](https://github.com/user-attachments/assets/cf7c74dd-9207-4695-a0b1-f98172a03069)